### ISSUE TITLE
Raw sql query is way slower on Android than iOS

### ISSUE THREAD

**Issue by mattermoran** [tekartik/sqflite#202]
Running sql query with order takes about 27 seconds while on iOS it only takes 34ms. Why that would happen?
Here's the repo you could use to see. https://github.com/mattermoran/flutter-sqflite-test

**Comment by alextekartik** [comment]
Thanks for taking the time in creating a repo. Indeed I get the same result (have not tested on iOS yet though but it definitely super slow on Android).

I have tried the same query on sqlitebrowser on ubuntu and I get even worse result:
```
30 rows returned in 32380ms from: SELECT u.* FROM users u LEFT JOIN user_tags ut ON u.id = ut.user_id WHERE ut.tag_id in (3) ORDER BY ut.rank DESC LIMIT 30 OFFSET 0;
```

So maybe something is wrong with the database itself...or the query. It seems here adding the order screw the timing. Personnally I like creating views for querying data. I can impose the optimization I want (in your case here the ordering). For example you can try that. It looks far faster.

```
DROP VIEW IF EXISTS user_tag_view;
CREATE VIEW user_tag_view AS SELECT u.*, ut.rank as rank, ut.tag_id as tag_id FROM users u, user_tags ut WHERE u.id = ut.user_id ORDER BY ut.rank DESC;
SELECT u.id, u.name, u.rank FROM user_tag_view u WHERE u.tag_id in (3) LIMIT 30 OFFSET 0;
```
Should be about this in dart code:
```dart
 var batch = db.batch()
      ..execute('DROP VIEW IF EXISTS user_tag_view')
      ..execute(
          'CREATE VIEW user_tag_view AS SELECT u.*, ut.rank as rank, ut.tag_id as tag_id FROM users u, user_tags ut WHERE u.id = ut.user_id ORDER BY ut.rank DESC');
    await batch.commit();
    query =
        'SELECT u.id, u.name, u.rank FROM user_tag_view u WHERE u.tag_id in (3) LIMIT 30 OFFSET 0';
```

Here I get the result in 3ms on Android

**Comment by mattermoran** [comment]
@alextekartik thanks a lot. Seems like using views makes it waaay faster.

**Comment by kevinengel** [comment]
I've also observed a large performance difference between Android and iOS.

Within a single database transaction I issue O(5000) database instructions (rawInsert, rawUpdate, rawQuery etc). On Android this takes ~ 20s, on iOS, it takes ~ 2s.

Is this factor of 10 performance gap expected? Is there anything I can do to speed up Android? As per the recommendation above, I tried removing all of the ORDER BYs from my queries but this didn't make much of a difference.
